load/load_distributor.go

```diff
@@ -1,12 +1,44 @@
 package load
 
-import "fmt"
+import (
+	"fmt"
+
+	consumer "github.com/AdityaMayukhSom/ruskin/consumer"
+)
+
+// Making a map of TopicId and LoadPartition
+// The map values have to be mapped to LoadPartition uid(will implement later)
+var TopicToPartitionMap = make(map[string]*LoadPartition)
 
 type LoadDistributor struct {
 	PartitionsCount int
 	Partitions      []LoadPartition
 }
 
+func (ld *LoadDistributor) getOrCreatePartition(topic string) *LoadPartition {
+
+	//Checking if loadPartition exist in the map
+	if partition, exist := TopicToPartitionMap[topic]; exist {
```

Comment:
This access should be thread safe as this can be spawned between multiple goroutines for faster servicing. Use RWMutex and use RLock while reading and Lock while adding new partition.